File: 프로그래머스/unrated/138476. 귤 고르기/귤 고르기.py
def solution(k, tangerine):
    dic = {}
    for t in tangerine:
        if t not in dic:
            dic[t] = 1
        else:
            dic[t] += 1
    data = [value for key, value in dic.items()]
    data.sort(reverse=True)
    
    
    result = 0
    i = 0
    while k > result:
        result += data[i]
        i += 1
    
    return i
        

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.java
import java.util.Comparator;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.stream.Collectors;

class Solution {
    public int solution(int k, int[] tangerine) {
        Map<Integer, Integer> dic = new HashMap<>();
        for (int t : tangerine) {
            dic.put(t, dic.getOrDefault(t, 0) + 1);
        }

        List<Integer> sizes = dic.values().stream()
                .sorted(Comparator.reverseOrder())
                .collect(Collectors.toList());

        int result = 0;
        for (Integer size : sizes) {
            k -= size;
            result += 1;

            if (k <= 0) {
                break;
            }
        }

        return result;
    }
}